structuredheadings/dialogs/selectstyle.js

```diff
@@ -0,0 +1,35 @@
+CKEDITOR.dialog.add("selectStyle", function (editor) {
+
+  var styleOptions = [["Default", null]];
+
+  //Build out the style options
+  (function () {
+    for (var style in CKEDITOR.config.autonumberStyles) {
+      styleOptions.push([style, style]);
```

Comment:
First part: I don't think order ultimately matters here. It's only building a radio select and allowing the user to choose which they want. I could run it through .sort first? BUT I'm also not iterating over the list so I'm not clear if the order is still affected. I think the problem only shows itself in a for-in loop, othwerwise the numeric index order is maintained. So I think we are good here.

Second Part: In my sample, I'm never touching config, the plugin is adding these values itself, so they will never be undefined. I actually need to test that you can specify your own without the plugin overwriting with it's defaults. I'm thinking currently only the defaults will be available.